scipy/spatial/ckdtree.pyx

```diff
@@ -1035,24 +1060,45 @@ cdef public class cKDTree [object ckdtree, type ckdtree_type]:
         else:
             raise ValueError("Invalid output type") 
 
+    @cython.boundscheck(False)
+    def build_weights(cKDTree self, object weights):
+        cdef np.intp_t num_of_nodes
+        cdef np.ndarray[np.float64_t, ndim=1, mode="c"] node_weights;
+        cdef np.ndarray[np.float64_t, ndim=1, mode="c"] _weights
+
+        num_of_nodes = self.tree_buffer.size();
+        node_weights = np.empty(num_of_nodes, dtype=np.float64)
+        _weights = np.ascontiguousarray(weights, dtype=np.float64)
+
+        if len(_weights) != self.n:
+            raise ValueError('Number of weights differ from the number of data points')
+
+        build_weights(<ckdtree*> self, <np.float64_t*>np.PyArray_DATA(node_weights),
+                            <np.float64_t*> np.PyArray_DATA(weights))
+                    
+        return node_weights
 
     # ---------------
     # count_neighbors
     # ---------------
 
     @cython.boundscheck(False)
-    def count_neighbors(cKDTree self, cKDTree other, object r, np.float64_t p=2.):
+    def count_neighbors(cKDTree self, cKDTree other, object r, np.float64_t p=2., 
+                        object self_weights=None, object other_weights=None, int cumulative=True):
         """
-        count_neighbors(self, other, r, p=2.)
+        count_neighbors(self, other, r, p=2., self_weights=None, other_weights=None, cumulative=True)
```

Comment:
Here is the thoughts I went through:

- 'weights' may confuse people when 'other' and 'self' are the same tree -- does it refer to weight on both or just on 'self'?

- 'self_weights' and 'other_weights' are too many typing and just looks very patchy;

- I considered an alternative where the tree carries the weights (such that the count_neighbors signature doesn't need to be changed), but decided against it -- it is nicer being able to reuse the same tree for different set of weights;

So I don't particularly like self_weights and other_weights neither. Any suggestions?